python-phoenixdb/phoenixdb/avatica/client.py

```diff
@@ -173,7 +174,7 @@ def _post_request(self, body, headers):
                 requestArgs.update(verify=self.verify)
```

Comment:
Wouldn't performing the authentication setup  on the Session object in open_connection() once per sessions be better?

We could also pre-set the content-type on the Session object, and wouldn't have to pass the headers parameter around.

The same could be done for the stream flag.

https://docs.python-requests.org/en/master/user/advanced/#session-objects
https://2.python-requests.org/en/master/api/#request-sessions

Reply:
I would guess that we wouldn't see a big improvement while we are suffering from CALCITE-4152 (spnego re-negotiation), however I agree with you.

I'll try to implement your suggestions. Thanks!